components/article-footer/feedback.js

```diff
@@ -0,0 +1,156 @@
+import { LitElement, html } from "lit";
+
+import { L10nMixin } from "../../l10n/mixin";
+
+import styles from "./feedback.css?lit";
+
+/**
+ * @typedef {"outdated"|"incomplete"|"code_examples"|"technical"|"consistency"|"incomprehensible"|"linguistic"|"other"} FeedbackReason
+ */
+
+/** @type {Partial<Record<FeedbackReason, string>>} */
+const FEEDBACK_REASONS = {
+  outdated: "Content is out of date",
+  incomplete: "Missing information",
+  code_examples: "Code examples not working as expected",
+  other: "Other",
+};
+
+/** @type {Partial<Record<FeedbackReason, string>>} */
+const FEEDBACK_REASONS_DE = {
+  technical: "Übersetzung enthält fachliche Fehler",
+  consistency: "Begriffe sind inkonsistent übersetzt",
+  incomprehensible: "Übersetzung ist nicht verständlich",
+  linguistic: "Übersetzung enthält sprachliche Fehler",
+  code_examples: "Code-Beispiele funktionieren nicht",
+  other: "Sonstige",
+};
+
+export class ArticleFooterFeedback extends L10nMixin(LitElement) {
+  static styles = styles;
+
+  static properties = {
+    locale: { type: String },
+    _reason: { state: true },
+    _view: { state: true },
+  };
+
+  constructor() {
+    super();
+    this.locale = "";
+    this._reason = "";
+    /** @type {'vote'|'feedback'|'thanks'} */
+    this._view = "vote";
+  }
+
+  /**
+   * @param {boolean} value
+   */
+  _handleVote(value) {
+    this._view = value ? "thanks" : "feedback";
+    // Reusing Thumbs' key to be able to reuse queries.
+    // FIXME gleanClick(`${THUMBS}: ${ARTICLE_FOOTER} -> ${Number(value)}`);
+  }
+
+  _handleFeedback() {
+    this._view = "thanks";
+    // FIXME gleanClick(`${ARTICLE_FOOTER}: feedback -> ${reason}`);
+  }
+
+  _renderVote() {
+    return html`<label
+        >${this.l10n(
+          "article_footer_feedback_question",
+        )`Was this page helpful to you?`}
+      </label>
+      <div class="button-container">
+        <button
+          type="button"
+          class="button yes"
+          @click=${{ handleEvent: () => this._handleVote(true) }}
+        >
+          ${this.l10n`Yes`}
+        </button>
+        <button
+          type="button"
+          class="button no"
+          @click=${{ handleEvent: () => this._handleVote(false) }}
+        >
+          ${this.l10n`No`}
+        </button>
```

Comment:
@LeoMcA Do you have a tip why these click handlers don't attach?

See: https://fred-pr45.review.mdn.allizom.net/en-US/docs/Web#article_footer

Reply:
Thanks, fixed via 44f9d67.